packages/client/lib/sync/fetcher/headerfetcher.ts

```diff
@@ -11,7 +11,7 @@ export interface HeaderFetcherOptions extends BlockFetcherOptions {
  * Implements an les/1 based header fetcher
  * @memberof module:sync/fetcher
  */
-export class HeaderFetcher extends BlockFetcher {
+export class HeaderFetcher extends Fetcher {
```

Comment:
Yep this makes a lot of sense, I did something similar in #1023, it doesn't make a lot of sense that `HeaderFetcher` builds upon `BlockFetcher`. (In fact, it overrides all methods except `tasks`)